blockly-core/core/block_svg_functional.js

```diff
@@ -68,9 +70,19 @@ Blockly.BlockSvgFunctional.prototype.createFunctionalMarkers_ = function () {
       continue;
     }
     this.inputMarkers_[input.name] = Blockly.createSvgElement('rect', {
-      fill: 'red' // todo
+      fill: 'red'
     }, this.svgGroup_);
   }
+
+  // Remove input markers that disappeared
+  Object.keys(this.inputMarkers_).forEach(function (markerName) {
```

Comment:
Nice, thanks for doing this!

It _feels_ like there's a fancy chaining way of doing this, but since it's a remove, AND do something based on the removed element, not sure it'd be any more readable or concise